### PR TITLE
Feature/claim aave comp to user

### PR DESCRIPTION
**Pull request by dumebi**
Claim aave and comp to user if loan status is `completed` or `liquidated`. If not, claim accrued interest to escrow address

### CODE REVIEW

**Comment by admazzola** [comment]
yarn test polygon gives me: 

18 passing (2m)
  1 failing

  1) Story Test Suite - DAPP.CLAIM.AAVE
       AAVE Claim AAVE:
     Error: VM Exception while processing transaction: reverted with reason string '29'


Do you see the same ?

**Comment by dumebi** [comment]
> yarn test polygon gives me:
> 
> 18 passing (2m)
> 1 failing
> 
> 1. Story Test Suite - DAPP.CLAIM.AAVE
>    AAVE Claim AAVE:
>    Error: VM Exception while processing transaction: reverted with reason string '29'
> 
> Do you see the same ?

yes, same. 

On the issue

**Review by mkhoury93 — APPROVED** [review]

File: contracts/escrow/dapps/AaveClaimAaveFacet.sol [hunk 1]
```diff
@@ -53,29 +63,27 @@ contract AaveClaimAaveFacet is PausableMods, DappMods {
...
-                msg.sender,
-                amount
-            )
+            abi.encodeWithSelector(IStakedAave.redeem.selector, user, amount)
         );
 
         // Add AAVE to escrow token list
         // TODO: get the AAVE token (add as immutable constructor variable)
-        // LibEscrow.tokenUpdated(loanID, address(0));
+        LibEscrow.tokenUpdated(loanID, address(0));
```
Comment (passabilities) [inline contracts/escrow/dapps/AaveClaimAaveFacet.sol:86]: 1. this should fail when called as this calls `balanceOf` on the token address passed in. 
2. should only call this if the escrow contract claims the reward (if user's address gets the reward, no need to update the escrow token array)

File: contracts/escrow/dapps/CompoundClaimCompFacet.sol [hunk 2]
```diff
@@ -43,10 +46,12 @@ contract CompoundClaimCompFacet is PausableMods, DappMods {
...
+        address user = LibLoans.loan(loanID).status >= LoanStatus.Closed
+            ? msg.sender
+            : address(LibEscrow.e(loanID));
         LibEscrow.e(loanID).callDapp(
             address(COMPTROLLER_ADDRESS_PROVIDER_ADDRESS),
-            abi.encodeWithSignature("claimComp(address)", escrow)
+            abi.encodeWithSignature("claimComp(address)", user)
         );
 
         // Add AAVE to escrow token list
```
Comment (passabilities) [inline contracts/escrow/dapps/CompoundClaimCompFacet.sol:57]: same comment from Aave

**Review by passabilities — CHANGES_REQUESTED** [review]

File: contracts/escrow/dapps/AaveClaimAaveFacet.sol (hunk 1 above)
Comment (dumebi) [reply contracts/escrow/dapps/AaveClaimAaveFacet.sol:86]: will do

File: contracts/escrow/dapps/CompoundClaimCompFacet.sol (hunk 2 above)
Comment (dumebi) [reply contracts/escrow/dapps/CompoundClaimCompFacet.sol:57]: done

File: contracts/escrow/dapps/AaveClaimAaveFacet.sol [hunk 3]
```diff
@@ -53,29 +63,31 @@ contract AaveClaimAaveFacet is PausableMods, DappMods {
         uint256 amount,
         address[] calldata tokenAddresses
```
Comment (passabilities) [inline contracts/escrow/dapps/AaveClaimAaveFacet.sol:64]: lets add `tokenAddresses` to the comment block above. are these supposed to be the underlying token addresses (DAI or USDT) or the tokens we are claiming (AAVE, aDAI)?

File: contracts/escrow/dapps/CompoundClaimCompFacet.sol [hunk 4]
```diff
@@ -43,15 +46,19 @@ contract CompoundClaimCompFacet is PausableMods, DappMods {
...
             address(COMPTROLLER_ADDRESS_PROVIDER_ADDRESS),
-            abi.encodeWithSignature("claimComp(address)", escrow)
+            abi.encodeWithSignature("claimComp(address)", user)
         );
 
-        // Add AAVE to escrow token list
-        // TODO: get the COMP token (add as immutable constructor variable)
-        LibEscrow.tokenUpdated(loanID, address(0));
+        if (loanUnavailable) {
+            LibEscrow.tokenUpdated(loanID, LibLoans.loan(loanID).lendingToken);
```
Comment (passabilities) [inline contracts/escrow/dapps/CompoundClaimCompFacet.sol:60]: we should rather be calling this function with the token addresses that we are updating (i.e. in this case `COMP`)

see my **TODO** comment that was removed in the diff above

File: test/helpers/story/drivers/dapp-utils/aave.utils.ts [hunk 5]
```diff
@@ -134,12 +138,21 @@ export const aaveClaimTest = async (
...
-  const aaveBefore = await IncentiveController.getUserUnclaimedRewards(
-    escrowAddress
-  )
+
+  shouldPass = await hre.evm.withBlockScope(0, async () => {
+    // do the mint for the deployer
+    const borrowerAddress = await borrower.getAddress()
+    await hre.evm.advanceTime(moment.duration(1, 'day'))
+    const assets = [details.lendingToken.address]
+    console.log({ assets, borrowerAddress })
```
Comment (passabilities) [inline test/helpers/story/drivers/dapp-utils/aave.utils.ts:147]: be sure to remove any `console.log`s

**Review by passabilities — CHANGES_REQUESTED** [review]

File: test/helpers/story/drivers/dapp-utils/aave.utils.ts (hunk 5 above)
Comment (dumebi) [reply test/helpers/story/drivers/dapp-utils/aave.utils.ts:147]: switch to hre.log? or completely remove

File: contracts/escrow/dapps/AaveClaimAaveFacet.sol (hunk 3 above)
Comment (dumebi) [reply contracts/escrow/dapps/AaveClaimAaveFacet.sol:65]: done

**Pull request merged**